spinny_gun.py

```diff
@@ -76,6 +76,60 @@ def intersects(rect, radius, center):
     return corner_distance_sq <= radius ** 2.0
 
 
+def unpause():
+    global pause
+    pygame.mixer.music.unpause()
+    pause = False
+
+
+def paused():
+
+    pygame.mixer.music.pause()
+
+    pause_surf_1, pause_rect_1 = text_objects(
+        "PAUSED",
+        MEDIUM_TEXT,
+        WHITE,
+        (DISPLAY_WIDTH * 0.5, DISPLAY_HEIGHT * 0.375),
+    )
+
+    pause_instructions_surf_1, pause_instructions_rect_1 = text_objects(
+        "Press 'p' to resume",
+        MEDIUM_TEXT,
+        WHITE,
+        (DISPLAY_WIDTH * 0.5, DISPLAY_HEIGHT * 0.55),
+    )
+
+    pause_instructions_surf_2, pause_instructions_rect_2 = text_objects(
+        "Press 'q' to quit",
+        MEDIUM_TEXT,
+        WHITE,
+        (DISPLAY_WIDTH * 0.5, DISPLAY_HEIGHT * 0.65),
+    )
+
+    while pause:
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                pygame.quit()
+                quit()
+            if event.type == pygame.KEYUP:
+                if event.key == pygame.K_p:
+                    unpause()
+            if event.type == pygame.KEYUP:
+                if event.key == pygame.K_q:
+                    pygame.quit()
+                    quit()
```

Comment:
```suggestion
                    exit_game()
```

Should probably use the same exit game function throughout, in case we want to attach anything to it later.